doc/writing_new_controller.rst

```diff
@@ -0,0 +1,138 @@
+
+   2. include ``"controller_interface/controller_interface.hpp"`` and ``visibility_control.h`` if you are using one.
+
+   3. Define namespace for your controller. This is usually a package name written in ``snake_case``.
+
+   4. Define the class of the controller, extending ``ControllerInterface``, e.g.,
+      .. code:: c++
+         class ControllerName : public controller_interface::ControllerInterface
+
+   5. Add constructor without parameters and the following public methods overriding ``ControllerInterface`` definition: ``init``, ``command_interface_configuration``, ``state_interface_configuration``, ``on_configure``, ``on_activate``, ``on_deactivate``, ``update``.
+      For exact definitions check the ``controller_interface/controller_interface.hpp`` header or one of the controller from `ros2_controllers <https://github.com/ros-controls/ros2_controllers> `_.
+
+   6. (optional) Often, controllers accept lists of joint names and interface names as parameters.
+      If so, you can add two protected string vectors where to store those values.
+
+4. **Adding definitions into source file (.cpp)**
+
+   1. Include the header file of your controller and add a namespace definition to simplify further development.
+
+   2. (optional) Implement constructor if needed. There, you could initialize member variables.
+      This could also be done in the ``init`` method.
+
+   3. Implement the ``init`` method. The first line usually calls for the parent ``init`` method.
+      Here is the best place to initialize the variables would be best if you initialize all the variables, reserve memory, and most important, declare node parameters used by the controller. If everything works fine return ``controller_interface::return_type::OK`` or ``controller_interface::return_type::ERROR`` otherwise.
+
+   4. Write the ``on_configure`` method. Parameters are usually read here, and everything is prepared so that controller can be started.
+
+   5. Implement ``command_interface_configuration`` and ``state_interface_configuration`` where required interfaces are defined.
+      There are three options of the interface configuration ``ALL``, ``INDIVIDUAL``, and ``NONE`` defined in ``controller_interface/controller_interface.hpp"``.
+      ``ALL`` and ``NONE`` option will ask for access to all available interfaces or none of it. The ``INDIVIDUAL`` configuration needs a detailed list of required interface names. Those are usually provided as parameters.
+      The full interface names have structure ``<joint_name>/<interface_type>``.
+
+   6. Implement the ``on_activate`` method with checking, and potentially sorting, the interface and assigning members initial values.
+      This method is part of the real-time loop, therefore avoid any reservation of memory and, in general, keep it as short as possible.
+
+
+   7. Implement the ``on_deactivate`` method, which does the opposite of ``on_activate``.
```

Comment:
```suggestion
   7. Implement the ``on_deactivate`` method, which does the opposite of ``on_activate``.
```

Reply:
I can't see the diff on this nor can github apply it... I'll consider it a glitch but please feel free to open a PR to fix this if you know what the original intention here was.